psyneulink/library/compositions/pytorchcomponents.py

```diff
@@ -37,11 +38,49 @@ def get_fct_param_value(param_name):
         leak = get_fct_param_value('leak')
         return lambda x: (torch.max(input=(x - bias), other=torch.tensor([0], device=device).double()) * gain +
                             torch.min(input=(x - bias), other=torch.tensor([0], device=device).double()) * leak)
-
     else:
         raise Exception(f"Function {function} is not currently supported in AutodiffCompositions!")
 
-class PytorchMechanismWrapper():
+def wrap_mechanism(mechanism, index, device, context=None):
+    return PytorchMechanismWrapper(mechanism, index, device, context=context)
+
+class PytorchWrapper():
+    def _get_learnable_param_ids(self):
+        return []
+
+    def _get_learnable_param_struct_type(self, ctx):
+        structs = []
+        for param_id in self._get_learnable_param_ids():
+            param = getattr(self._object.parameters, param_id)
+            assert param is not None, f"Failed to get param {param_id} from {self}"
```

Comment:
This assert message is deceptive. `gettattr` raises an exception if the attribute is present.
iiuc, `parameters.?` will get an instance of `Parameter` class and cannot be `None`.